### PR TITLE
[persistence] optimize first heartbeat

### PR DESCRIPTION
**Pull request by cyrilou242**
See https://startree.atlassian.net/browse/TE-2586

We can skip the first heartbeat by setting the `lastActive` value at acquisition directly. 
this saves 2 queries for all task that run faster than the heartbeat frequency. 

For most detection, this means the number of queries per task is reduced by 2.
At high scale this matters. The heartbeat queries take locks.

Note:
the heartbeat logic is still bad because it has a concurrency issue but this will be taken in another PR. See https://startree.atlassian.net/browse/TE-2578

### CODE REVIEW

File: thirdeye-persistence/src/main/java/ai/startree/thirdeye/datalayer/bao/TaskManagerImpl.java [hunk 1]
```diff
@@ -147,13 +148,15 @@ AND ref_id not in (select ref_id from task_entity where status = 'RUNNING')
    * This method has side effects on the task DTO, even if the acquisition attempt fails.
    * Re-fetch the taskDto if you need to ensure consistency with the persistence layer.
    */
+  @Deprecated
```
Comment (anshul98ks123) [inline thirdeye-persistence/src/main/java/ai/startree/thirdeye/datalayer/bao/TaskManagerImpl.java:151]: why are we deprecating this method?

Comment (cyrilou242) [reply thirdeye-persistence/src/main/java/ai/startree/thirdeye/datalayer/bao/TaskManagerImpl.java:151]: it was already deprecated in the interface, it's only used by the legacy task acquisition logic
same for the other one

**Review by anshul98ks123 — APPROVED** [review]
:shipit:

File: thirdeye-persistence/src/main/java/ai/startree/thirdeye/datalayer/bao/TaskManagerImpl.java (hunk 1 above)
Comment (cyrilou242) [reply thirdeye-persistence/src/main/java/ai/startree/thirdeye/datalayer/bao/TaskManagerImpl.java:151]: `acquireNextTaskToRun`
should be used instead, it performs find and acquire in a single optimized way

**Pull request merged**